server.go

```diff
@@ -1121,7 +1136,14 @@ func (s *server) InboundPeerConnected(conn net.Conn) {
 	// Check to see if we should drop our connection, if not, then we'll
 	// close out this connection with the remote peer. This
 	// prevents us from having duplicate connections, or none.
-	if connectedPeer, ok := s.peersByPub[pubStr]; ok {
+	connectedPeer, err := s.findPeerByPubStr(pubStr)
+	switch err {
+	case ErrPeerNotFound:
+		// We were unable to locate an existing connection with the
+		// target peer, proceed to connect.
+		break
```

Comment:
Not sure if breaks like this are used throughout the codebase.

Reply:
Yeah thinking we should just ditch the switch all together. Working on fix now

Reply:
Well, not sure if it gets more readable. Checking  `err == ErrPeerNotFound` makes it very easy to quickly glance why we are just continuing in case of a returned error.

```go
if connectedPeer, err := s.findPeer(peerStr); err == nil {
  // Peer is already connected
  ...do stuff ...
} else if err != ErrPeerNotFound {
    log(err)
    return
}
s.peerConnected()
```
Even though the last clause never will happen, it makes it easier to see that we sometime actually expects an error.

Reply:
The `findPeerByPubStr` is just meant to wrap the internal map and provide a concrete error type, so it should have the same semantics as `peer, ok := s.peersByPub[pubStr]`, where `ok` can only have two possible values. Since `nil` and `ErrPeerNotFound` are the only two errors returned, would it be more readable to just check `err != ErrPeerNotFound`? I'd rather not introduce code that is unreachable by construction. If enumerating the errors helps with readability, then maybe using the switch was the better alternative?

Reply:
Meant to answer this before :P Agree that not introducing unreachable code probably is a good idea. Doing the `err != ErrPeerNotFound` should be readable enough, if accompanied by a comment :)

Reply:
Looks like you reverted back to the `switch`, which also looks good to me (but would appreciate a comment explaining why no default case). Your call 🎺

Reply:
@halseth Thanks for the quick turn around! Agree that a comment would be useful here, I added one indicating that we only expect to handle the two possible error values :)

Reply:
LGTM!